packages/contracts-bedrock/scripts/deploy/AddGameType.s.sol

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+// Forge
+import { Script } from "forge-std/Script.sol";
+
+// Scripts
+import { DeployUtils } from "scripts/libraries/DeployUtils.sol";
+
+// Interfaces
+import { OPContractsManager } from "src/L1/OPContractsManager.sol";
+import { ISystemConfig } from "interfaces/L1/ISystemConfig.sol";
+import { IProxyAdmin } from "interfaces/universal/IProxyAdmin.sol";
+import { IDelayedWETH } from "interfaces/dispute/IDelayedWETH.sol";
+import { IBigStepper } from "interfaces/dispute/IBigStepper.sol";
+import { GameType, Duration, Claim } from "src/dispute/lib/Types.sol";
+import { IFaultDisputeGame } from "interfaces/dispute/IFaultDisputeGame.sol";
+
+/// @title AddGameType
+contract AddGameType is Script {
+    struct Input {
+        // Address that will be used for the DummyCaller contract
+        address prank;
+        // OPCM contract address
+        OPContractsManager opcmImpl;
+        // SystemConfig contract address
+        ISystemConfig systemConfigProxy;
+        // ProxyAdmin contract address
+        IProxyAdmin opChainProxyAdmin;
+        // DelayedWETH contract address (optional)
+        IDelayedWETH delayedWETHProxy;
+        // Game type to add
+        GameType disputeGameType;
+        // Absolute prestate for the game
+        Claim disputeAbsolutePrestate;
+        // Maximum game depth
+        uint256 disputeMaxGameDepth;
+        // Split depth for the game
+        uint256 disputeSplitDepth;
+        // Clock extension duration
+        Duration disputeClockExtension;
+        // Maximum clock duration
+        Duration disputeMaxClockDuration;
+        // Initial bond amount
+        uint256 initialBond;
+        // VM contract address
+        IBigStepper vm;
+        // Whether this is a permissioned game
+        bool permissioned;
+        // Salt mixer for deterministic addresses
+        string saltMixer;
+    }
+
+    struct Output {
+        IDelayedWETH delayedWETHProxy;
+        IFaultDisputeGame faultDisputeGameProxy;
+    }
+
+    function run(Input memory _agi) public returns (Output memory ago_) {
+        addGameType(_agi, ago_);
+        checkOutput(ago_);
+        return ago_;
+    }
+
+    function addGameType(Input memory _agi, Output memory _ago) internal {
```

Comment:
this is a weird pattern. It looks like `run()` is declaring an empty `ago_` return value, then passing it to `addGameType()` which populates it.

A more typical pattern would be something like: 

```suggestion
    function run(Input memory _agi) public returns (Output memory) {
        Output memory ago_ = addGameType(_agi);
        checkOutput(ago_);
        return ago_;
    }

    function addGameType(Input memory _agi) internal returns (Output memory) {
```